src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,445 @@
+              style={{
+                display: this.showValue ? "" : "none",
+                height:
+                  this.minorSize * 0.795 -
+                  this.calcThickness() * (this.minorSize / 100) +
+                  "px",
+                transform:
+                  calcPercentage(this) >= 100
+                    ? "rotate(359deg)"
+                    : calcPercentage(this) > 0
+                      ? "rotate(" + 3.6 * calcPercentage(this) + "deg)"
+                      : "rotate(0.5deg)"
+              }}
+            >
+              <div
+                class="indicator"
+                style={{
+                  "box-shadow":
+                    (this.minorSize <= 300 && this.thickness <= 25) ||
+                    !this.styleShadow
+                      ? "none"
+                      : "",
+                  height:
+                    this.minorSize / 100 +
+                    this.calcThickness() * (this.minorSize / 150) +
+                    "px",
+                  transform:
+                    "translateY(-" +
+                    (this.minorSize / 100 +
+                      this.calcThickness() * (this.minorSize / 150)) +
+                    "px)"
+                }}
+              />
+            </span>
+          ) : (
+            ""
+          )}
+          <div
+            class="gauge"
+            style={{
+              "background-color": this.styleCenterColor,
+              "box-shadow":
+                (this.minorSize <= 300 && this.thickness <= 25) ||
+                !this.styleShadow
+                  ? "none"
+                  : "",
+              height:
+                this.minorSize * 0.792 -
+                this.calcThickness() * (this.minorSize / 100) +
+                "px",
+              width:
+                this.minorSize * 0.792 -
+                this.calcThickness() * (this.minorSize / 100) +
+                "px"
+            }}
+          >
+            {this.showValue ? (
+              <div
+                style={{
+                  color: "" + this.styleCenterTextColor,
+                  "font-size":
+                    (this.minorSize * 0.795 -
+                      this.calcThickness() / 2 * (this.minorSize / 100)) /
+                      8 +
+                    "px",
+                  "mix-blend-mode": "difference"
+                }}
+              >
+                {(calcPercentage(this) > 100
+                  ? 100
+                  : calcPercentage(this) < 0
+                    ? 0
+                    : Math.round(calcPercentage(this))) + "%"}
+              </div>
+            ) : (
+              ""
+            )}
+          </div>
+        </div>
+      );
+    } else if (this.gaugeType === "Line") {
+      const divRanges = [];
+      const divRangesName = [];
+      let currentMargin = 0;
+      let currentMargin2 = 0;
+      //////////////////////////////////////////
+      function calcPositionRange(preValue) {
```

Comment:
Not sure if it justifies having a function to increment a counter. Maybe just use `currentMargin += preValue;`